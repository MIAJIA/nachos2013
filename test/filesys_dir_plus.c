#include "filesysgrader.h"
#include "stdio.h"
#include "stdlib.h"

char buf[100];

int main(int argc, char** argv)
{
    int size = getFreeDiskSize();
    chdir("..");
    chdir(".");
	//Assert that rmdir can't be done
    assertTrue(rmdir("nachos"));
    assertTrue(!mkdir("nachos"));
    chdir("./nachos/./../nachos/.");
    assertTrue(!rmdir("../nachos"));
    mkdir("/svn");
    mkdir("../svn/./././acm.sjtu.edu.cn");
    chdir("./.././svn/././acm.sjtu.edu.cn/./.");
    printCwd();
    chdir(".././nachos/./../tenet/../.././../");
    printCwd();
    assertTrue(rmdir("/svn"));
    chdir("./..");
    assertTrue(!rmdir("/svn/acm.sjtu.edu.cn"));
    assertTrue(!rmdir("/svn"));
    assertTrue(mkdir("jns"));
	chdir("/");
    assertTrue(getFreeDiskSize() == size);
    done();
    return 0;
}
