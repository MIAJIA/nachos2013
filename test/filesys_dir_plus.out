/svn/acm.sjtu.edu.cn
/svn/acm.sjtu.edu.cn
